prql-compiler/src/semantic/eval/mod.rs

```diff
@@ -0,0 +1,558 @@
+    for index in 0..len {
+        let mut row = Vec::new();
+        for field in &fields {
+            row.push(Expr {
+                alias: field.alias.clone(),
+                ..field.kind.as_array().unwrap()[index].clone()
+            })
+        }
+
+        rows.push(Expr::from(ExprKind::Tuple(row)));
+    }
+
+    Ok(Expr::from(ExprKind::Array(rows)))
+}
+
+fn std_module() -> Expr {
+    Expr::from(ExprKind::Tuple(
+        [
+            new_func("floor", &["x"]),
+            new_func("add", &["x", "y"]),
+            new_func("neg", &["x"]),
+            new_func("select", &["closure", "relation"]),
+            new_func("derive", &["closure", "relation"]),
+            new_func("filter", &["closure", "relation"]),
+            new_func("aggregate", &["closure", "relation"]),
+            new_func("window", &["closure", "relation"]),
+            new_func("columnar", &["closure", "relation"]),
+            new_func("sum", &["x"]),
+            new_func("lag", &["x"]),
+        ]
+        .to_vec(),
+    ))
+}
+
+fn new_func(name: &str, params: &[&str]) -> Expr {
+    let params = params
+        .iter()
+        .map(|name| FuncParam {
+            name: name.to_string(),
+            default_value: None,
+            ty: None,
+        })
+        .collect();
+
+    let kind = ExprKind::Func(Box::new(Func {
+        name_hint: Some(Ident {
+            path: vec!["std".to_string()],
+            name: name.to_string(),
+        }),
+
+        // these don't matter
+        return_ty: Default::default(),
+        body: Box::new(Expr::null()),
+        params,
+        named_params: Default::default(),
+        args: Default::default(),
+        env: Default::default(),
+    }));
+    Expr {
+        alias: Some(name.to_string()),
+        ..Expr::from(kind)
+    }
+}
+
+fn zip_relations(l: Expr, r: Expr) -> ExprKind {
+    let l = l.kind.into_array().unwrap();
+    let r = r.kind.into_array().unwrap();
+
+    let mut res = Vec::new();
+    for (l, r) in zip(l, r) {
+        let l_fields = l.kind.into_tuple().unwrap();
+        let r_fields = r.kind.into_tuple().unwrap();
+
+        res.push(Expr::from(ExprKind::Tuple([l_fields, r_fields].concat())));
+    }
+
+    ExprKind::Array(res)
+}
+
+#[cfg(test)]
+mod test {
+
+    use insta::assert_display_snapshot;
+    use itertools::Itertools;
+
+    use super::*;
+
+    fn eval(source: &str) -> Result<String> {
+        let stmts = crate::prql_to_pl(source)?.into_iter().exactly_one()?;
+        let expr = stmts.kind.into_var_def()?.value;
+
+        let value = super::eval(*expr)?;
+
+        Ok(value.to_string())
+    }
+
+    #[test]
+    fn basic() {
+        assert_display_snapshot!(eval(r"
+            [std.floor (3.5 + 2.9) + 3, 3]
+        ").unwrap(),
+            @"[9, 3]"
+        );
+    }
+
+    #[test]
+    fn tuples() {
+        assert_display_snapshot!(eval(r"
+              {{a_a = 4, a_b = false}, b = 2.1 + 3.6, c = [false, true, false]}
```

Comment:
I didn't know we supported nested types yet!

Reply:
Well, the compiler does not support it, but you can parse them :D

Which is enough for the evaluator.